### ISSUE TITLE
Ability to hack on an already replaced module

### ISSUE THREAD

**Issue by myitcv** [rogpeppe/gohack#8]
I _think_ this makes sense; based on an actual scenario. 

A module `M` I was working on had a dependency on `D1`. But I'd already forked `D1` to `D1'`. Hence my `go.mod` already had a replace from `D1 => D1'`.

So my desire to hack on `D1` translates to needing to hack on `D1'`. And hence `gohack` should be following replace directives to work out what should ultimately be hacked on. Instead I got:

```
"github.com/shurcooL/vfsgen" is already replaced; will not override replace statement in go.mod
all modules failed; not replacing anything
error: [
        {/home/myitcv/gostuff/src/gopkg.in/errgo.v2/fmt/errors/alias.go:15: all modules failed; not replacing anything}
]
```

**Comment by myitcv** [comment]
This gets a little tricky because "replacing the replace" is, in effect, lossy. So we will have to work out how to "push" the original replace state.

**Comment by rogpeppe** [comment]
Do you want to change a replace statement that's referring to a directory, or one that refers to another module?

**Comment by rogpeppe** [comment]
fixed by #24
